# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/samanashfaq05/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [3]:
!git clone https://github.com/samanashfaq05/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 155, done.
remote: Counting objects: 100% (155/155), done.
remote: Compressing objects: 100% (111/111), done.
remote: Total 155 (delta 62), reused 94 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (155/155), 1.86 MiB | 12.10 MiB/s, done.
Resolving deltas: 100% (62/62), done.


In [4]:
%cd /content/flyrank-ml-internship

/content/flyrank-ml-internship


In [5]:
import os

print(os.getcwd())
print(os.path.exists("data/raw/content_refresh_anonymized.csv"))

/content/flyrank-ml-internship
True


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Content Refresh Prioritization

I chose the Content Refresh Prioritization lane because websites often contain thousands of pages, but only a small portion need immediate attention. Reviewing every page manually is time-consuming, so I want to use machine learning to help identify pages that are more likely to experience declining search performance.

This lane matches the work I completed in the starter notebooks, where I explored declining content, compared a hand-written rule with a decision tree, and built my first model using engineered features. My goal is not to replace human decisions but to provide a ranked list that helps content teams decide which pages to review first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

Which content pages are most likely to experience declining search performance and should be prioritized for refresh?

### Decision

The model helps decide which content pages should be reviewed first.

### Action

The SEO or content team reviews the recommended pages and updates the content by improving quality, adding new information, fixing outdated sections, or optimizing SEO elements.

### Cost of a Wrong Decision

If the model recommends a healthy page (false positive), the team may spend time updating content that did not need attention.

If the model misses a truly declining page (false negative), that page may continue losing search visibility and traffic before anyone notices.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

declining = (df["trend_direction"].str.lower() == "down").sum()
declining_rate = (df["trend_direction"].str.lower() == "down").mean()

print("Total Pages:", len(df))
print("Declining Pages:", declining)
print("Declining Rate:", round(declining_rate,3))

Total Pages: 30000
Declining Pages: 16262
Declining Rate: 0.542


### Initial observations

The starter dataset contains **30,000** content pages.

Approximately **54.2%** of the pages are labeled as declining.

In the full warehouse notebook, more than **111,000** content items had enough history to build features for modeling. These numbers suggest that content decline is common enough to justify building a prioritization system instead of reviewing every page manually.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project does not prove why Google rankings change, nor can it guarantee that refreshing a page will improve search performance. The model only learns patterns from historical data and provides decision support.

The recommendations should be treated as a prioritized review list rather than automatic decisions. Human expertise is still required before updating or publishing content.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.